v1/services/storage/series_cursor.go

```diff
@@ -38,6 +38,18 @@ type indexSeriesCursor struct {
 }
 
 func newIndexSeriesCursor(ctx context.Context, predicate *datatypes.Predicate, shards []*tsdb.Shard) (*indexSeriesCursor, error) {
+	var expr influxql.Expr
+	if root := predicate.GetRoot(); root != nil {
+		var err error
+		if expr, err = reads.NodeToExpr(root, measurementRemap); err != nil {
+			return nil, err
+		}
+	}
+
+	return newIndexSeriesCursorInfluxQL(ctx, expr, shards)
```

Comment:
The name may not be ideal (cursor for influxql?) but the rest is good.

Reply:
How about `newIndexSeriesCursorInfluxQLPred`?

Reply:
sounds good